content/blog/2018/06/2018-06-08-jenkins-java10-hackathon.adoc

```diff
@@ -0,0 +1,82 @@
+---
+:layout: post
+:title: "Jenkins & Java 10+ Online Hackathon (Jun 18-22)"
+:tags:
+- events
+- community
+- developer
+:author: oleg_nenashev
+---
+
+image:/images/logos/formal_java/256.png[Jenkins Java, role=center, float=right]
+
+On behalf of the Jenkins Events Team,
+I would like to invite you to the “Jenkins & Java 10 Online Hackathon” which will take place from June 18th to 22nd.
+We will be working together on Jenkins core and plugins in order
+to find and fix compatibility issues, share experience and have some fun.
+Everybody is welcome to join, independently of their Jenkins experience and amount of time they have available.
+
+If you are interested in participating in the hackathon, please sign-up in
+link:https://docs.google.com/forms/d/1ReYyuyCGC0PIz2quh6XehnjpH2K52inx-veHLPlNreE/edit[this form].
+
+### Background
+
+Java 9 has recently been end-of-lifed, Java 10 in GA, and Java 11 is in early beta.
+Jenkins project currently requires Java 8 to run reliably,
+and there are some known compatibility issues with higher Java versions.
+
+During the Jenkins World 2017 Hackathon,
+link:https://github.com/markewaite[Mark Waite] and
+link:https://github.com/batmat[Baptiste Mathus] spent some time exploring Java 9 compatibility in Jenkins.
+We are currently tracking compatibility issues in the
+link:https://issues.jenkins-ci.org/browse/JENKINS-40689[JENKINS-40689] EPIC,
+but there are likely many unknown issues in Jenkins core, plugins and in libraries we use in the project.
+We would like to continue their effort and work on Java 10+ support.
+
+### Objectives and Scope
+
+As I have said above,
+the goals are to explore/fix compatibility issues, share experience and have fun.
+We **DO NOT** plan to make Jenkins fully compatible with Java 10+
+during the hackathon,
+but we will try to integrate fixes and make them available.
```

Comment:
I think there needs to be some more preparation in this regard. `remoting` probably needs to be incrementalified, we need [CI tools](https://github.com/jenkins-infra/documentation/blob/master/ci.adoc#tool-installers) to include JDK 10, etc. The goal I would suggest for preparation is that we have infrastructure set up to show _that_ stuff is completely broken currently on JDK 9+ and _how_ it is broken, so that participants can jump in with PRs offering fixes and see right away if they work, and collaborate on fixes.

Reply:
That would be fantastic if we could have those things prepared, but I don't know that I would consider them blockers. I suspect we can still get value from this exercise without these items, but any of them that can be in place will smooth the operation.

Reply:
I'd guess only a fraction of people will be working on Remoting, so we can workaround it by a branch and timestamped snapshots if needed